src/main/java/uk/gov/justice/digital/client/dynamodb/DynamoDBClient.java

```diff
@@ -25,57 +25,73 @@
 @Singleton
 public class DynamoDBClient {
 
-    private final static ObjectMapper mapper = new ObjectMapper();
     private static final Logger logger = LoggerFactory.getLogger(DynamoDBClient.class);
 
     private final static String indexName = "secondaryId-type-index";
     private final static String sortKeyName = "secondaryId";
 
     private final AmazonDynamoDB dynamoDB;
+    ObjectMapper mapper;
 
     @Inject
     public DynamoDBClient(JobParameters jobParameters) {
-        dynamoDB = AmazonDynamoDBClientBuilder.standard()
+        this(AmazonDynamoDBClientBuilder.standard()
                 .withEndpointConfiguration(new AwsClientBuilder.EndpointConfiguration(
                         jobParameters.getAwsDynamoDBEndpointUrl(),
                         jobParameters.getAwsRegion()
                 ))
-                .build();
+                .build(), new ObjectMapper());
+    }
+
+    public DynamoDBClient(AmazonDynamoDB dynamoDB, ObjectMapper mapper) {
+        this.dynamoDB = dynamoDB;
+        this.mapper = mapper;
     }
 
     public DomainDefinition getDomainDefinition(final String domainTableName, final String domainId)
             throws PatternSyntaxException {
 
         String[] names = domainId.split("[.]");
-        String domainName = names.length >= 2 ?names[0] :domainId;
+        String domainName = names.length == 2 ? names[0] : domainId;
```

Comment:
Perhaps we should tighten things up here. Does it make sense to have more or less than 2 terms in the array? I suspect not. Likewise for the tableName - I don't think it makes sense for the table name to be null does it?

If not then it's appropriate to throw an exception here since we should not continue unless we have exactly two terms.

Reply:
1 or 2 fields max. second one is optional. More than 2 not possible. for example domain_name.table_name in domain files. If table_name not provided then we do the action of insert/update/delete for all tables. I need to clarify this with Dom anyway because, my understanding as per latest document is, we never do any operation at domain level and it always be handled by the job which trigger the domain full refresh (for example lambda)

Reply:
Done as discussed